File: src/cocoa/logging.m
#include "../logging_native.h"

void debug(const char *fmt, ...) {
    if (utox_verbosity() < VERBOSITY_DEBUG) {
        return;
    }
    va_list l;
    va_start(l, fmt);
    NSLogv(@(fmt), l);
    va_end(l);
}

void debug_info(const char *fmt, ...) {
    if (utox_verbosity() < VERBOSITY_INFO) {
        return;
    }
    va_list l;
    va_start(l, fmt);
    NSLogv(@(fmt), l);
    va_end(l);
}

void debug_notice(const char *fmt, ...) {
    if (utox_verbosity() < VERBOSITY_NOTICE) {
        return;
    }
    va_list l;
    va_start(l, fmt);
    NSLogv(@(fmt), l);
    va_end(l);
}

void debug_error(const char *fmt, ...) {
    if (utox_verbosity() < VERBOSITY_ERROR) {
        return;
    }
    va_list l;
    va_start(l, fmt);
    NSLogv(@(fmt), l);
    va_end(l);
}

File: src/logging.c
#include "debug.h"

#include "settings.h"

#include <stdio.h>
#include <stdarg.h>
#include <string.h>

int utox_verbosity() {
    return settings.verbose;
}

void debug(const char *fmt, ...){
    if (!settings.debug_file) {
        return;
    }

    va_list list;
    va_start(list, fmt);
    vfprintf(settings.debug_file, fmt, list);
    va_end(list);

    printf("\n\n");

    #ifdef __WIN32__
    fflush(settings.debug_file);
    #endif
}
